server/controllers/nlp.js

```diff
@@ -47,12 +250,70 @@ exports.handleInput = function (req, res) {
 
         let dataset = req.body.dataset;
 
-        // Use NLP to find the right command
-        let command = findCommand(dataset, req.body.input);
+        let lowerCaseInput = req.body.input.toLowerCase();
+
+        var tokenizer = new natural.WordTokenizer();
+        let tokenizedInput = tokenizer.tokenize(lowerCaseInput);
+
+
+        let initState = {
+            tokens: tokenizedInput,
+            layer: 0,
+            currentToken: 0,
+            callback: state => {
+                callbackFunction(state, res)},
+            dataset: dataset
+        };
+
+        findCommand(initState);
+
         // Query the needed data and transform them into the dataformat for plotly.js
-        functions[command.function](dataset, command.parameters, data =>
-            // Send the data back to the client
-            res.send({data: data})
-        );
+
     })
-};
\ No newline at end of file
+};
+
+function callbackFunction(state, res){
```

Comment:
rename this to `findFunction` (maybe even `findDataTransformationFunction`, but this should match the list where these functions are in)